watertap/unit_models/zero_order/uv_aop_zo.py

```diff
@@ -0,0 +1,130 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+This module contains a zero-order representation of a UV-A0P unit.
+operation.
+"""
+
+from pyomo.environ import units as pyunits, Var
+from pyomo.common.config import ConfigValue, In
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core import declare_process_block_class
+from watertap.core import build_siso, constant_intensity, ZeroOrderBaseData
+
+# Some more information about this module
+__author__ = "Adam Atia"
+
+
+@declare_process_block_class("UVAOPZO")
+class UVAOPZOData(ZeroOrderBaseData):
+    """
+    Zero-Order model for a UV-AOP unit operation.
+    """
+
+    CONFIG = ZeroOrderBaseData.CONFIG()
+
+    CONFIG.declare("has_aop", ConfigValue(
+            default=False,
+            domain=In([True, False]),
+            description="Oxidant construction flag",
+            doc="""Indicates whether oxidant dose and flow terms should be constructed or not.
+            **default** - False."""))
```

Comment:
Looking at the code below, are we sure we shouldn't be just using inheritance for this? E.g., the class `UVZO` would be all the code with `has_aop == False`, and the class `UVAOPZO` would inherit from it and built the additional Vars/Constraints in its build.

It would also seem you wouldn't need the over-written `load_parameters_from_database` in either class, simplifying the code.

Reply:
Now that I think about it, this could work nicely because `UVAOPZO` would inherit the same tech type name of "uv_aop" and point to the same yaml file. I guess the main reason I did all this was to merge those two yaml files (uv_irradiation and uv_aop) into a single uv_aop, which would include both uv alone and uv/aop.

Reply:
You could do that, although I would be cautious.

The other thing to note is that this might be a case for supporting multiple subtypes in a single unit (but that has its own issues).

Reply:
With that said, although using inheritance would result in cleaner code, is that something we think I should spend time on to refine when the code (seems to) works fine as is? In my mind, it might be better to make only the subtle adjustments suggested and move on to the numerous other models that we have to merge. What are your thoughts @andrewlee94 and @bknueven ?

Reply:
Given @andrewlee94 's response elsewhere in this PR, I'll proceed with separating the two and using inheritance. It shouldn't take long.